### PR TITLE
[ARCHIVE] generate fields with a draftJS content state on save

### PR DESCRIPTION
**Pull request by jerome-poisson**
SDESK-4911

### CODE REVIEW

**Comment by jerome-poisson** [comment]
we are lacking a sequence diagram for the Superdesk operation. I believe generating during the `save` is enough, but is there any use case when saving is not used?

**Comment by jerome-poisson** [comment]
@tomaskikutis suggested that we add a check for a little while to check that client and backend generation do match, cf. https://dev.sourcefabric.org/browse/SDESK-4911?focusedCommentId=119033&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-119033

**Comment by petrjasek** [comment]
I think we might need to call this also when saving correction

**Comment by tomaskikutis** [comment]
We have that websocket notification when any resource changes, how about using that or a similar generic method instead of coupling this to features?

**Comment by jerome-poisson** [comment]
I've added the safety check, generation on correction, and the tests should be fixed.

File: superdesk/editor_utils.py [hunk 1]
```diff
@@ -282,7 +293,18 @@ def render(self):
                 del blocks[-1]
             html = self.exporter.render(content_state)
         else:
-            html = self.exporter.render(self.content_state)
+            try:
+                html = self.exporter.render(self.content_state)
+            except KeyError as e:
+                if e.args == ('text',):
+                    # "text" may be missing in some case (e.g. comments), and the exporter
+                    # doesn't support it. To avoid a crash, be render again after
```
Comment (petrjasek) [inline superdesk/editor_utils.py:301]: `s/be/we/`

File: features/archive.feature [hunk 2]
```diff
@@ -1116,3 +1116,62 @@ Feature: News Items Archive
...
+        Given "archive"
+        """
+        [{"_id": "test_editor_gen_1", "guid": "test_editor_gen_1", "headline": "test"}]
+        """
+
+        When we patch given
+        """
+        {
+            "fields_meta": {
+                "body_html": {
```
Comment (petrjasek) [inline features/archive.feature:1131]: can you also add headline to tests? I wonder how it handles single line fields

File: superdesk/editor_utils.py (hunk 1 above)
Comment (jerome-poisson) [reply superdesk/editor_utils.py:301]: good catch, thanks

File: features/archive.feature (hunk 2 above)
Comment (jerome-poisson) [reply features/archive.feature:1131]: sure

Comment (jerome-poisson) [reply features/archive.feature:1131]: @petrjasek done

**Comment by jerome-poisson** [comment]
@ride90 @petrjasek after testing again, I see that I have many inconsistencies between client and backend, notably the headline is text only with client generated (even if I  add bold and underline), while it's HTML with the backend. I'm not sure how to check it, investigating.

**Comment by tomaskikutis** [comment]
I think it's hardcoded here https://github.com/superdesk/superdesk-client-core/blob/develop/scripts/apps/authoring/views/article-edit.html#L42

To fix it properly we should move that property to a content profile

**Comment by jerome-poisson** [comment]
oh right, thanks @tomaskikutis , I'm putting `WIP` label on this PR until this is fixed.
@petrjasek @ride90 what do you think? Should I do a quick fix with hardcoded values in a first time, and create a ticket to do it properly?

For the record, here is what I get with a headline with underline and bold:
```
WARNING:superdesk.editor_utils:Generated HTML inconsistency between client and backend:
client field: Grève du 16 janvier
backend field: <p><b><u>Grève</u></b> du 16 janvier</p>
field state: {'blocks': [{'key': 'nmcg', 'text': 'Grève du 16 janvier', 'type': 'unstyled', 'depth': 0, 'inlineStyleRanges': [{'offset': 0, 'length': 5, 'style': 'BOLD'}, {'offset': 0, 'length': 5, 'style': 'UNDERLINE'}], 'entityRanges': [], 'data': {'MULTIPLE_HIGHLIGHTS': {}}}], 'entityMap': {}}

```

**Comment by jerome-poisson** [comment]
I have also inconsistency with special char like  non breakable space which is put verbatim by the backend, when `&nbsp;` is used by the client. Should be easy to fix.

**Comment by jerome-poisson** [comment]
I've updated the PR with:
- hardcoding of text only values (`headline` and `slugline`), I've created https://dev.sourcefabric.org/browse/SDESK-5316 to implement a cleaner solution later
- remove warning for COMMENTS, there is no rendering of comments, so we just ignore them
- strip text only fields, to be consistent with client
- use client version when the is inconsistency between client and backend.

For the latter, and as I've said this morning, the goal is to inspect the log during the transition period, without breaking existing generation, but I'm not sure if it's the right thing to do, maybe it's better to go directly with backend version to catch problems early? Also I've used a warning, but an error would catch more attention.

There is still the non-breakable space issue, client generate a `&nbsp;` while in backend it's a simple space. I've inspected and the `nbsp` is not present in draftJS state received in the backend. After checking with @tomaskikutis, it seems that client replaces `nbsp` by common space, but this need more investigation. I'll put this in a separate ticket for now.

 @ride90 and @petrjasek I know that you have already validated this PR, but could you double check it with the modification before I merge? Thanks!

**Comment by jerome-poisson** [comment]
**note**: here is the ticket about the NBSP issue ==> https://dev.sourcefabric.org/browse/SDESK-5320

**Pull request merged**